src/main/java/com/shoppinglist/shoppinglist/ShoppingListController.java

```diff
@@ -36,10 +39,15 @@ public String index(Model model) {
      * Save an item and redirect to the root endpoint.
      */
     @PostMapping("/save")
-    public String save(ListItem item) {
+    public String saveItem(ListItem item) {
         try {
             createListItem(item);
+        } catch (NoSuchElementException e) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, e.getMessage(), e);
+        } catch (IllegalArgumentException e) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, e.getMessage(), e);

```

Comment:
I separated the exception handling block into different blocks so that I can specify how each exception type should be handled.
https://rhamedy.medium.com/a-short-summary-of-java-coding-best-practices-31283d0167d3